### PR TITLE
New package: v2ray-4.39.2

### PR DESCRIPTION
**Pull request by ipkalm**
#### General
- [x] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

**Comment by noarchwastaken** [comment]
It seems like you are using the `nobody` user here. It seems tempting to use since it's the least privileged user in the system, but the consequence is that you get access to other processes running as `nobody`, such as NFS.

https://askubuntu.com/questions/329714/what-is-the-purpose-of-the-nobody-user

I think we should add a dedicated user for this package.

File: srcpkgs/v2ray/files/v2ray/run [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+exec chpst -u nobody:nogroup v2ray -config=/etc/v2ray/config.json 2>&1
```
Comment (noarchwastaken) [inline srcpkgs/v2ray/files/v2ray/run:3]:
```suggestion
exec chpst -u v2ray:v2ray v2ray -config=/etc/v2ray/config.json 2>&1
```

File: srcpkgs/v2ray/template [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+build_style=go
+go_import_path="github.com/v2fly/v2ray-core/v4"
+short_desc="Platform for building proxies to bypass network restrictions"
+maintainer="ipkalm <ipkalm@outlool.com>"
+license="MIT"
+homepage="https://github.com/v2fly/v2ray-core"
+distfiles="https://github.com/v2fly/v2ray-core/archive/v${version}/${pkgname}-${version}.tar.gz"
+checksum=b06d2bf63fd08d8a92d51824f69e41345bd90d747d444ca37770b2017a2a5a00
+conf_files="/etc/v2ray/config.json"
+
```
Comment (noarchwastaken) [inline srcpkgs/v2ray/template:15]:
```suggestion

system_accounts="v2ray"

```

**Review by noarchwastaken — CHANGES_REQUESTED** [review]

File: srcpkgs/v2ray/files/v2ray/run (hunk 1 above)
Comment (ericonr) [reply srcpkgs/v2ray/files/v2ray/run:3]: Accounts provided by packages should be prefixed with `_`: `_v2ray`.

**Comment by ipkalm** [comment]
@noarchwastaken I'll try do my best. This is 2nd PR in my life. Hope for understanding :blush:

**Comment by noarchwastaken** [comment]
wait, I just looked at your template again and it seems like you are specifying the `do_build` yourself...

You don't need to do that. The whole point of the templates is to simplify the templates; same as `do_check`. Those are usually managed by `xbps-src`.

Check out other Go packages like `caddy` for some examples.

**Comment by ipkalm** [comment]
@noarchwastaken ok, thank you.

**Comment by ipkalm** [comment]
Well. Now it is single commit! I did it :)

About my template file: `do_check` function doesn't provided in `common/build-style/go.sh`. So I wrote it into template. And because v2ray project uses `github.com/v2fly/v2ray-core/main` for `v2ray` binary and `github.com/v2fly/v2ray-core/infra/control/main` for `v2ctl` binary, `go install` command from `common/build-style/go.sh` build all this binary files with name `main` and put into the same directory. That is why I added `do_build()` func into `template` file. As example I used kubernetes, caddy, prometheus, docker templates from `srcpkgs` directory.

Hope you understand my English :grimacing:

**Comment by noarchwastaken** [comment]
> To build this binaries with specific names, then put them into `${GOPATH}/bin` directory. And `do_isntall` func from `common/build-style/go.sh` could install this files. As example I used kubernetes, caddy, prometheus, docker templates from `srcpkgs` directory.

FYI, you can also move those binaries in `post_build`. For multiple main's in one repo, I guess yours is the way to go.

> Hope you understand my English grimacing

I'm a newcomer here too! I only have a few PRs and just stumbled upon yours because I use v2ray in China as well.

**Comment by ipkalm** [comment]
> FYI, you can also move those binaries in `post_build`. For multiple main's in one repo, I guess yours is the way to go.

Do you mean move in `post_build` without my custom `do_build`?

**Comment by ipkalm** [comment]
> Yes. But, as you said, it's a multi-main repo, I'm not sure if it would build at all without the custom do_build.

I think it impossible.
In freebsd ports I saw this:
```
GO_TARGET=	./main:v2ray \
		./infra/control/main:v2ctl
```  
For each package we can specify name for output files. But I didn't find similar feature for xbps template.

**Comment by ipkalm** [comment]
@ericonr hi. Can you check please my commit? Thank you.

**Comment by ericonr** [comment]
Try taking a look at the `go_package` variable to build two binaries without a custom `do_build`.

**Comment by ipkalm** [comment]
@ericonr nice, thank you 😊

**Comment by ipkalm** [comment]
Sorry about reopen. I tried cherry-pick this PR commit from <my fork>-<master> branch to <my fork>-<v2ray> branch... But I couldn't. 😬

**Comment by ipkalm** [comment]
@ericonr greetings. Aare there any other improvements or changes required from me?

**Comment by abenson** [comment]
Why did you close this?  You just need to rebase on master to get rid of the merge commit.

**Comment by ipkalm** [comment]
@abenson sorry, but I did many mistakes while work with fork (most big is create changes in master branch, not in separate branch). I promise to be more careful for the next time.

**Pull request closed without merging**